google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelectionDialog.java

```diff
@@ -215,7 +217,12 @@ public void onFailure(Throwable throwable) {
   private void refreshDialog() {
     CredentialedUser currentUser = (CredentialedUser) accountComboBox.getSelectedItem();
     loadAllProjects();
-    accountComboBox.setSelectedItem(currentUser);
+    if (currentUser != null) {
+      accountComboBox.setSelectedItem(currentUser);
+    } else {
+      // no selection before, select active user by default.
```

Comment:
Did you confirm that the most recently logged in user will be the active user?

The behavior I think we are after is that the _most recently logged in user_ is the one selected in the dropdown, regardless of if its the default user or not.

Reply:
So, when we reload, we take the currently selected user, and after reload, set it back as selected so the state is preserved. If nothing was selected (no signed in users first), we check for null and set user to active - which I think is the most recently logged in user.

Reply:
Yeah thats what I wanted to be sure of - that the most recently logged in user will always be selected (even if I am already logged in to one or more accounts). I think, that be adding a new account I'd want that new account selected on successful login (even if there are other accounts already logged in).

Reply:
Got it, updated - so manual refresh now keeps the selection, but if we change accounts, i.e. add new active one, dialog will switch to new active one and load its project list.